### ISSUE TITLE
Add JSDoc for plugins in Getting Started

### ISSUE THREAD

**Issue by RafaelGSS** [fastify/fastify#3725]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### Documentation update

I was talking with @mcollina and looks good to add JSDocs to our guides in the getting started section. It will enable autocompletion in their code editor and make things more beginner-friendly.

File: https://github.com/fastify/fastify/blob/main/docs/Guides/Getting-Started.md#your-first-plugin

**Comment by RafaelGSS** [comment]
Example:

```diff
// our-first-route.js

+ /**  @type {import('fastify').FastifyPluginAsync<{}>} */
async function routes (fastify, options) {
  fastify.get('/', async (request, reply) => {
    return { hello: 'world' }
  })
}

module.exports = routes
```

**Comment by karansapolia** [comment]
I'd like to work on this. I haven't added JSDocs to projects before and feel like this could be a good way to get up to speed with it and also make a good first contribution to Fastify. With a little guidance, I think I can get this done :)

**Comment by Fdawgs** [comment]
How detailed are we wanting it to be @RafaelGSS?

As an example, I use JSDoc tags everywhere in my plugins/APIs:
https://github.com/Fdawgs/ydh-community-contacts-api/blob/src/plugins/db/index.js#L7-L15

**Comment by RafaelGSS** [comment]
> How detailed are we wanting it to be @RafaelGSS?

Actually, I meant `type doc` https://jsdoc.app/tags-type.html. It will enable autocompletion.

> I'd like to work on this. I haven't added JSDocs to projects before and feel like this could be a good way to get up to speed with it and also make a good first contribution to Fastify. With a little guidance, I think I can get this done :)

Go ahead! If you are stuck, create a DRAFT PR so we can help you.

**Comment by Sarfraz-droid** [comment]
Hi! can i work on this?

**Comment by mcollina** [comment]
go for it!

**Comment by climba03003** [comment]
Closed in favor of #3850
